code-challenges.test.js

```diff
@@ -21,36 +20,161 @@
+const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"];
 
-// a) Create a test with expect statements for each of the variables provided.
+const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"];
+
+const oddOnly = (array) => {
+  let oddTime = array.filter(
+    (value) => typeof value === "number" && value % 2 !== 0
+  );
+  oddTime.sort((a, b) => a - b);
+  return oddTime;
+  // I would like feedback on how the => works on line 90 please
+};
+
+console.log(oddOnly(fullArr1));
+console.log(oddOnly(fullArr2));
+
+// function oddOnly(array){
+//   if (array % 2 !== 0) {
+//     // array.filter
+//     console.log(oddOnly);
+//   }
+
+// const filter = array.filter((number) => number % 2)
+
+// function oddOnly(number) {
+//   if (typeof number === "number") {
+//     return (oddOnly);
+//   }
+// }
+
+// // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+
+// // a) Create a test with expect statements for each of the variables provided.
 
-const numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+describe("arraySum", () => {
+  it("takes in an array and returns an array of the accumulating sum. An empty array should return an empty array", () => {
+    const numbersToAdd1 = [2, 4, 45, 9];
+    // Expected output: [2, 6, 51, 60]
 
-const numbersToAdd2 = [0, 7, -8, 12]
+    const numbersToAdd2 = [0, 7, -8, 12];
+    // Expected output: [0, 7, -1, 11]
+
+    const numbersToAdd3 = [];
+    // Expected output: []
+
+    expect(arraySum(numbersToAdd1)).toEqual([2, 6, 51, 60]);
+    expect(arraySum(numbersToAdd2)).toEqual([0, 7, -1, 11]);
+    expect(arraySum(numbersToAdd3)).toEqual([]);
+  });
+});
+
+// // The best failing possible
+
+// // b) Create the function that makes the test pass.
+
+// // PS:
+// // create arraySum function
+// // takes in array
+// // create empty array
+// // add each index to the next (including previous values)
+// // i[0] + i[1], i[0] + i[1] + i[2] etc
+// // .reduce(previous value, current value, index) method
+// // call function
+
+const numbersToAdd1 = [2, 4, 45, 9];
+// Expected output: [2, 6, 51, 60]
+
+const numbersToAdd2 = [0, 7, -8, 12];
 // Expected output: [0, 7, -1, 11]
 
-const numbersToAdd3 = []
+const numbersToAdd3 = [];
 // Expected output: []
 
-
-// b) Create the function that makes the test pass.
+// My first attempt; close, but doesn't return all the values and doesn't put them in an array
+
+// function arraySum(array) {
+//   let array2 = [];
+//   return array.reduce(
+//     (previous, current, i) => (array2[i] = previous + current),
+//     0
+//   );
+// }
+
+const arraySum = (array) => {
+  let emptyArray = [];
+  let addingSum = 0;
+  for (let i = 0; i < array.length; i++) {
+    addingSum = addingSum + array[i];
+    emptyArray.push(addingSum);
+  }
+  return emptyArray;
+};
+console.log(arraySum(numbersToAdd1))
```

Comment:
With the testing suites, you do not need to console.log or do function calls...because it is all taken care of by the test you created...the expect statement does it for you.

Reply:
Thank you for bringing that to my attention. I'll keep that in mind for next time!